x/builder/ante/ante.go

```diff
@@ -55,47 +53,51 @@ func (ad BuilderDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool,
 		}
 	}
 
-	isAuctionTx, err := ad.mempool.IsAuctionTx(tx)
+	bidInfo, err := ad.mempool.GetAuctionBidInfo(tx)
 	if err != nil {
 		return ctx, err
 	}
 
 	// Validate the auction bid if one exists.
-	if isAuctionTx {
+	if bidInfo != nil {
 		// Auction transactions must have a timeout set to a valid block height.
-		if err := ad.HasValidTimeout(ctx, tx); err != nil {
-			return ctx, err
-		}
-
-		bidInfo, err := ad.mempool.GetAuctionBidInfo(tx)
-		if err != nil {
-			return ctx, err
+		if int64(bidInfo.Timeout) < ctx.BlockHeight() {
+			return ctx, fmt.Errorf("timeout height cannot be less than the current block height")
 		}
 
-		// If the current transaction is the highest bidding transaction, then the highest bid is empty.
-		topBid := sdk.Coin{}
-
 		// We only need to verify the auction bid relative to the local validator's mempool if the mode
 		// is checkTx or recheckTx. Otherwise, the ABCI handlers (VerifyVoteExtension, ExtendVoteExtension, etc.)
 		// will always compare the auction bid to the highest bidding transaction in the mempool leading to
 		// poor liveness guarantees.
+		topBid := sdk.Coin{}
```

Comment:
by forcing bid to be an sdk.Coins() we get less generalizability. For instance in the polaris implementation, this forces us to pass `evmDenom` around, whereas if topBid could be just an uint64 for instance, then we could get rid of this simplifiying the integration.

Really the only thing that bids are required to be are `comparable`

Reply:
Agreed. Using `sdk.Coin` would make more sense in a world where we have multiple fee tokens. Opening an issue for this.